src/recognition/ReadIDX.java

```diff
@@ -0,0 +1,313 @@
+            } else {
+                path = path.concat(sc.nextLine() + "\\");
+            }
+        }
+        return path;
+    }
+
+
+
+
+    /********************************************************************************
+     *
+     * getFile Method got file from folder
+     *
+     * If file doesn't exists, method asking user try another path
+     *
+     *
+     * @param folderPath - path to the folder with files
+     * @param fileName - name of file that should be found
+     *
+     * @return file
+     *******************************************************************************/
+    File getFile(String folderPath, String fileName) {
+        File file = new File(folderPath + "\\"+fileName);
+
+        if ( file.isFile() ) {
+            System.out.println("File " + file.getName() + " was found");
+        } else {
+            System.out.println("File " + file.getName() + " wasn't found");
+            getImages(false, false);
+        }
+        return file;
+    }
+
+
+
+    /********************************************************************************
+     *
+     * updateCollection Method updates collection with read images
+     *
+     * It use two files - one with labels to images (1- for image with draw of one)
+     * the second one contains images which are represented like matrix 28*28 where
+     * every element is a gradation of gray in current pixel 0-white, 255 black.
+     *
+     * Method checks compatibility of this two files and their continuity
+     *
+     * @param labels - file with labels to numbers
+     * @param images - file with pixels description
+     * @param col - Collection that should be updated
+     *
+     * @return boolean result of update. True if OK.
+     *******************************************************************************/
+    boolean updateCollection(File labels, File images, ArrayList<ImageIDX> col) {
+
+        byte[] labelsData = new byte[0];
+        byte[] imagesData = new byte[0];
+        int labelsIndex = 8;
+        int imagesIndex = 16;
+
+        int labelsLength;
+        int imagesLength;
+
+
+        try ( FileInputStream stream = new FileInputStream(labels) ) {
+            labelsData = stream.readAllBytes();
+
+        } catch (IOException exc) {
+            System.out.println("\nProblem with file reading");
+        }
+
+        try ( FileInputStream stream = new FileInputStream(images) ) {
+            imagesData = stream.readAllBytes();
+
+        } catch (IOException exc) {
+            System.out.println("\nProblem with file reading");
+        }
+
+
+        labelsLength = convertBytesToInt(labelsData[4], labelsData[5], labelsData[6], labelsData[7]);
+        imagesLength = convertBytesToInt(imagesData[4], imagesData[5], imagesData[6], imagesData[7]);
+
+        /*
+        Checking that labels and images files are equal and corrrect
+        */
+        if ((labelsLength != imagesLength) || (labelsData[3] != 1) || (imagesData[3] != 3) ||
```

Comment:
It is common practice to encapsulate complex conditions in a special method that has a clear name like
```
if (something()) {
    // ....
}
```

Reply:
Thank you, I'll try to apply it.